dbms/src/Server/Client.cpp

```diff
@@ -445,12 +510,41 @@ class Client : public Poco::Util::Application
         return select(1, &fds, 0, 0, &timeout) == 1;
     }
 
+    const String prompt() const
+    {
+        String pattern = prompt_by_server_display_name;
+
+        for (const auto & [key, value]: colors)
+        {
+            String name = static_cast<String>(key);
+            String code = static_cast<String>(value);
+
+            boost::replace_all(pattern, "[" + name + "]",  "\33[1;" + code + "m");
+            boost::replace_all(pattern, "[/" + name + "]", "\033[0m");
+        }
+
+        std::map<String, String> environment = {
+            {"host",         config().getString("host", "localhost")},
+            {"port",         config().getString("port", "9000")},
+            {"display_name", server_display_name},
+            {"user",         config().getString("user",     "default")},
+            {"database",     config().getString("database", "default")},
+        };
+
+        for (const auto & [key, value]: environment)
+        {
+            boost::replace_all(pattern, "%" + static_cast<String>(key) + "%",  static_cast<String>(value));
+        }
+
+        return pattern;
+    }
 
     void loop()
     {
         String query;
         String prev_query;
-        while (char * line_ = readline(query.empty() ? ":) " : ":-] "))
+
+        while (char * line_ = readline(query.empty() ? prompt().c_str() : ":-] "))
```

Comment:
The function `prompt()` is called every time.

Reply:
Yes, because the user can change the current database and it must change in prompt. Maybe there is another way to do it?

Reply:
Every time, on client-side, once per request. Doesn't sound like a big issue, and i think that in that place avoiding that calls will be just premature optimization.